doc/howdoi.rst

```diff
@@ -11,6 +11,8 @@ How do I ...
 
    * - How do I...
      - Solution
+   * - add a DataArray to my dataset as a new variable
+     - ``my_dataset[varname] = my_dataArray``
```

Comment:
```suggestion
     - ``my_dataset[varname] = my_dataArray`` or :py:meth:`Dataset.assign`
```

We should also link to https://xarray.pydata.org/en/stable/data-structures.html#dictionary-like-methods. We have docs on building the documentation here: https://xarray.pydata.org/en/stable/contributing.html#contributing-to-the-documentation